### PR TITLE
[MM-28062] Add telemetry for in-app purchase flow

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Adds telemetry tracking for various places in the purchase flow, as described in: https://docs.google.com/document/d/102uElsng_cnpFHAr8tqTEIhktX5x5R9LAxDLBPDOvAU/edit#heading=h.y5i12eg5od68

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28062?filter=15333

#### Related Pull Requests
n/a
#### Screenshots
n/a

### CODE REVIEW

**Comment by nickmisasi** [comment]
@Adovenmuehle I've made the following assumption in order to make implementation easier. Let me know if you'd like me to change it:

`User clicks on ‘not right now’ on limit modal`  - I've also had this track when the user clicks the "x" button on the same modal, since they are both doing the same thing - closing the modal. 

`User clicks on ‘not right now’ on limit modal` - the event in the doc is listed as `click_modal_user_limit_updgrade` (note the spelling of `upgrade`). I've changed it to `click_modal_user_limit_upgrade`, I can update the doc for this.

**Review by ali-farooq0 — APPROVED** [review]
Nice work @nickmisasi!

File: components/purchase_modal/purchase_modal.tsx [hunk 1]
```diff
@@ -114,6 +117,12 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
                     <a
                         className='footer-text'
                         href='https://support.mattermost.com/hc/en-us/requests/new?ticket_form_id=360000640492'
+                        onClick={() =>
+                            trackEvent(
+                                TELEMETRY_CATEGORIES.CLOUD_PURCHASING,
+                                'click_contact _support',
```
Comment (Adovenmuehle) [inline components/purchase_modal/purchase_modal.tsx:123]: I think there's an erroneous space before `_support`

File: components/purchase_modal/purchase_modal.tsx [hunk 2]
```diff
@@ -178,6 +187,12 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
                     <a
                         className='footer-text'
                         href='https://support.mattermost.com/hc/en-us/requests/new?ticket_form_id=360000640492'
+                        onClick={() => {
+                            trackEvent(
+                                TELEMETRY_CATEGORIES.CLOUD_PURCHASING,
+                                'click_contact _sales',
```
Comment (Adovenmuehle) [inline components/purchase_modal/purchase_modal.tsx:193]: Same issue as above

File: components/user_limit_modal/user_limit_modal.tsx [hunk 3]
```diff
@@ -20,7 +22,18 @@ type Props = {
...
+        pageVisited(
+            TELEMETRY_CATEGORIES.CLOUD_PURCHASING,
+            'pageview_modal_user_limit_reached',
+        );
+    }, []);
+
     const onSubmit = () => {
+        trackEvent(
+            TELEMETRY_CATEGORIES.CLOUD_PURCHASING,
+            'click_modal_user_limit_updrade',
```
Comment (Adovenmuehle) [inline components/user_limit_modal/user_limit_modal.tsx:35]: Still typo on upgrade, could be losing my mind though

**Review by Adovenmuehle — CHANGES_REQUESTED** [review]
Couple of nits but otherwise LGTM. Agree with your changes mentioned in your comment.

File: components/user_limit_modal/user_limit_modal.tsx (hunk 3 above)
Comment (nickmisasi) [reply components/user_limit_modal/user_limit_modal.tsx]: Removed the g instead of the d. woops.

**Review by Adovenmuehle — APPROVED** [review]
🚢

**Comment by nickmisasi** [comment]
Unable to cherry-pick this properly until https://github.com/mattermost/customer-web-server/pull/245 is merged

**Pull request merged**